components/shapediver/parameter/select/SelectComponent.tsx

```diff
@@ -0,0 +1,76 @@
+import {MantineColor} from "@mantine/core";
+import React from "react";
+import SelectButtonFlexComponent from "./SelectButtonFlexComponent";
+import SelectButtonGroupComponent from "./SelectButtonGroupComponent";
+import SelectChipGroupComponent from "./SelectChipGroupComponent";
+import SelectColorComponent from "./SelectColorComponent";
+import SelectDropDownComponent from "./SelectDropDownComponent";
+import SelectFullWidthCards from "./SelectFullWidthCards";
+import SelectImageDropDownComponent from "./SelectImageDropDownComponent";
+
+export interface SelectComponentItemDataType {
+	/** Display name to use instead of the item name. */
+	displayname?: string;
+	/** Tooltip. */
+	tooltip?: string;
+	/** Description. */
+	description?: string;
+	/** URL to image. Can be a data URL including a base 64 encoded image. */
+	imageUrl?: string;
+	/** Optional color, used for color selection components. */
+	color?: MantineColor;
+	/** Optional width. */
+	width?: number;
```

Comment:
@LemToUp we could keep the width as an optional property here, but I guess typically one will want to configure this globally, see https://github.com/shapediver/AppBuilderShared/pull/77/files#r1992015404

Reply:
https://github.com/shapediver/AppBuilderShared/pull/77/commits/7c57aeba8fecb6c50269ca31ee057b7b409d8113